src/hooks/useKeyboardHandler.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback } from 'react'
+import { clearSelection, removeArrow, removeMarker } from '../reducer'
+import { useDispatch, useSelector } from '../store'
+import { redo, undo } from '../undoable'
+
+export default function useKeyboardUndoHandler() {
+  const dispatch = useDispatch()
+  const currentSelection = useSelector((state) => state.currentSelection)
+
+  const handler = useCallback(
+    (event: KeyboardEvent) => {
+      if (
```

Comment:
Mildly annoying that this is on 3 lines and the else on line 21 is on a single line. :smile:

Reply:
agreed. I'll extract `const isMetaOn = event.ctrlKey || event.metaKey`